src/test/java/org/jpeek/MetricsTest.java

```diff
@@ -55,229 +55,216 @@
-            new Object[] {"OnlyOneMethodWithParams", "LCC", 0.0d},
-            new Object[] {"OverloadMethods", "LCC", 1.0d},
-            new Object[] {"TwoCommonAttributes", "LCC", 0.0d},
-            new Object[] {"WithoutAttributes", "LCC", 0.0d},
-            new Object[] {"NoMethods", "CCM", Double.NaN},
-            new Object[] {"WithoutAttributes", "CCM", Double.NaN},
-            new Object[] {"OneMethodCreatesLambda", "CCM", Double.NaN},
-            new Object[] {"OneVoidMethodWithoutParams", "CCM", Double.NaN},
-            new Object[] {"Bar", "CCM", 0.125d},
-            new Object[] {"Foo", "CCM", 0.1667d},
-            new Object[] {"OverloadMethods", "CCM", 0.6d},
-            new Object[] {"TwoCommonAttributes", "CCM", Double.NaN},
-            new Object[] {"TwoCommonMethods", "CCM", 0.0238d},
-            new Object[] {"Bar", "MWE", 1.0d},
-            new Object[] {"Foo", "MWE", 1.0d},
-            new Object[] {"MethodMethodCalls", "MWE", 1.0d},
-            new Object[] {"MethodsWithDiffParamTypes", "MWE", 1.0d},
-            new Object[] {"NoMethods", "MWE", 1.0d},
-            new Object[] {"OneMethodCreatesLambda", "MWE", 1.0d},
-            new Object[] {"OneVoidMethodWithoutParams", "MWE", 1.0d},
-            new Object[] {"OnlyOneMethodWithParams", "MWE", 1.0d},
-            new Object[] {"OverloadMethods", "MWE", 1.0d},
-            new Object[] {"TwoCommonAttributes", "MWE", 1.0d},
-            new Object[] {"WithoutAttributes", "MWE", 1.0d},
-            new Object[] {"Foo", "CCM", 0.1667d},
-            new Object[] {"OverloadMethods", "CCM", 0.6d},
-            new Object[] {"TwoCommonAttributes", "CCM", Double.NaN},
-            new Object[] {"TwoCommonMethods", "CCM", 0.0238d},
-            new Object[] {"Bar", "OCC", 0.75d},
-            new Object[] {"BridgeMethod", "OCC", 0.0d},
-            new Object[] {"ClassWithPublicField", "OCC", 0.5d},
-            new Object[] {
-                "IndirectlyRelatedPairs",
-                "OCC",
-                0.6666666666666666d,
-            },
-            new Object[] {"MethodMethodCalls", "OCC", 0.2d},
-            new Object[] {
-                "MethodsWithDiffParamTypes",
-                "OCC",
-                0.3333333333333333d,
-            },
-            new Object[] {"NoMethods", "OCC", 0.0d},
-            new Object[] {"OneMethodCreatesLambda", "OCC", 0.0d},
-            new Object[] {"OneVoidMethodWithoutParams", "OCC", 0.0d},
-            new Object[] {"OnlyOneMethodWithParams", "OCC", 1d},
-            new Object[] {"OverloadMethods", "OCC", 0.75d},
-            new Object[] {"TwoCommonAttributes", "OCC", 0.0d},
-            new Object[] {"TwoCommonMethods", "OCC", 0.0d},
-            new Object[] {"WithoutAttributes", "OCC", 0.0d},
-            new Object[] {"TwoCommonMethods", "LORM", 0.26_667d}
-        );
-    }
-
-    @Test
-    public void testsTarget() throws Exception {
+    @ParameterizedTest
+    @MethodSource("targets")
```

Comment:
@HDouss Even though it is a feature of Junit5, I find this kind of features very confusing. Would personally not use.
@paulodamaso WDYT???

Reply:
@fanifieiev I think it still better than Junit 4 way where you return an `Object` array and then get mapped by indices to instance fields. Anyway, I think my last commit propose an approach that adresses all your comments.